aptos-move/framework/src/natives/string_utils.rs

```diff
@@ -174,13 +174,14 @@ fn native_format_impl(
             write!(out, "@{}", str).unwrap();
         },
         MoveTypeLayout::Signer => {
-            let addr = val.value_as::<move_core_types::account_address::AccountAddress>()?;
+            let strct = val.value_as::<Struct>()?;
+            let addr = strct.unpack()?.next().unwrap().value_as::<move_core_types::account_address::AccountAddress>()?;
             let str = if context.canonicalize {
                 addr.to_canonical_string()
             } else {
                 addr.to_hex_literal()
             };
-            write!(out, "signer({})", str).unwrap();
+            write!(out, "signer(@{})", str).unwrap();
```

Comment:
This is not backward compatible unfortunately. cc @lightmark @gerben-stavenga

Reply:
well it aborts so it;s not like this is emitted on-chain yet. But yes, abort vs non-abort is not technically a backwards compatible change

Reply:
abort can be unchain...
nvm, I guess the whole set of format has to be revisited but not only this one...

Reply:
@junkil-park have you run replay?

Reply:
It passed the replay test: https://github.com/aptos-labs/aptos-core/actions/runs/5756722358